### ISSUE TITLE
consolidate Diversity panel genome zoom state

### ISSUE THREAD

**Issue by eharkins** [nextstrain/auspice#1145]
**Current Behavior**  
In the redux state, we track the current zoom state of the diversity panel, which for example looks like this in the query `?gmax=4757&gmin=2379` and like this in the UI: 
![Screen Shot 2020-05-28 at 5 23 39 PM](https://user-images.githubusercontent.com/12140437/83207431-0a218280-a108-11ea-9b5b-ac24e3dd2a03.png)

To track this, we have both 
* `state.controls.zoomMin, state.controls.zoomMax`
* `state.entropy.zoomMin, state.entropy.zoomMax`. 

@jameshadfield and I agree that you should only need one of these two, since they do seem to be redundant (only the `state.controls` version is used, though perhaps it should be the other way around). James drew up this nice diagram to understand the current behavoir:

![Untitled-2020-05-29-1109](https://user-images.githubusercontent.com/8350992/83214127-eddd1e80-a1b7-11ea-8ef3-f43462b8c86d.png)

**Expected behavior**  
Just use one place in the redux state to track the zoom status of the Diversity panel.

**How to reproduce**  
Technically this works right now, hence the low priority and lack of bug tag. If we wanted to read this part of the state in some new code, however, we might easily write something that doesn't work by using the wrong part of the state.

**Possible solution**  
Replace use of `state.controls.zoomMin` and `state.controls.zoomMax` with the same fields from `state.entropy`, essentially collapsing the two boxes on the right of the diagram into one.

**Additional context**  
`absoluteZoomMin` and `absoluteZoomMax` are also [set here on `state.controls`](https://github.com/nextstrain/auspice/blob/master/src/actions/recomputeReduxState.js#L688-L689) and used in middleware to unset the url query when the zoom goes back to the ends.

Also important to also note from our conversation:
> It’s also complicated how you actually perform zooming in the d3 component. For instance, when you drag the brushes to zoom do you:
(a) dispatch an action, change redux state, then on a react lifecycle see that it’s changed and perform the imperative d3  command to zoom into the (new) value
(b) as you zoom change the d3 object as well as dispatching redux actions (or debounced dispatches, or only a single dispatch at the end). Then, when the react lifecycle updates (as redux state has changed) do you have a check to see that you don’t actually need to zoom in as it’d already been done in this case?

We do (b)
> which is why when dragging the brushes you only see the URL change when you release the brush

This should be able to stay that way as it seems like it should give better performance.

**Comment by jameshadfield** [comment]
Thanks @eharkins -- couple of other subtle bugs related to the redux state managment here:

* Zooming the entropy panel by the ctrl+mouseWheel does not update the URL query. Not sure what (if any) dispatches are occurring on this action
* App state is correctly reinstantiated via loading a page with URLs (e.g. [this one](https://nextstrain.org/zika?gmax=5656&gmin=5128)) but the URL subsequently looses the `gmax` & `gmin` queries.

**Comment by dos077** [comment]
The query is erased by a dispatch from EntropChart.prototype, defined in entropyD3.js:

![Annotation 2020-06-19 222929](https://user-images.githubusercontent.com/45016741/85189361-6b0e2780-b27c-11ea-82df-08ac0c02ac1f.jpg)

When I commented out the highlighted code the URL query stays. I don't understand the motive behind why this line was there in the first place.

**Comment by dos077** [comment]
After looking over the comment again, I think maybe the blanket else statement should be replace with an elseif, to make sure the zoom coords are not erase accidentally:

else if (
        d3event.sourceEvent.target.id.match(/^prot/) ||
        d3event.sourceEvent.target.id.match(/^nt/)
)

Should be what the comment is intended?

**Comment by eharkins** [comment]
Thanks @dos077 ! Reopening until I can break up this super-issue into sub-issues since your fix is only a piece of the puzzle here.

**Comment by eharkins** [comment]
@jameshadfield after #1175, the query remains after the entropy is toggled off. Is this what we want? If not, I think it would be easy enough to add some code to redux reducers and/or middleware to clear out things like this on the TOGGLE_PANEL_DISPLAY action that gets dispatched when toggling on/off panels.
